transmart-core-db/src/main/groovy/org/transmartproject/db/dataquery2/query/HibernateCriteriaQueryBuilder.groovy

```diff
@@ -252,7 +260,29 @@ class HibernateCriteriaQueryBuilder implements QueryBuilder<Criterion> {
      * Creates a criteria object for a patient set by conversion to a field constraint for the patient id field.
      */
     Criterion build(PatientSetConstraint constraint) {
-        build(new FieldConstraint(field: patientIdField, operator: Operator.CONTAINS, value: constraint.patientIds))
+
+        if (constraint.patientIds != null) {
+            build(new FieldConstraint(field: patientIdField, operator: Operator.IN, value: constraint.patientIds))
+        }
+        else if (constraint.patientSetId != null) {
+
+            QueryBuilder subQueryBuilder = new HibernateCriteriaQueryBuilder(
```

Comment:
I think, that a sub query builder is not required here, because it is not a subquery on observation fact, but on another table. You can call `getAlias` on the current builder (`this`).